shell/server/notifications-server.js

```diff
@@ -0,0 +1,128 @@
+// Sandstorm - Personal Cloud Sandbox
+// Copyright (c) 2016 Sandstorm Development Group, Inc. and contributors
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const Capnp = Npm.require("capnp");
+const SupervisorCapnp = Capnp.importSystem("sandstorm/supervisor.capnp");
+const SystemPersistent = SupervisorCapnp.SystemPersistent;
+
+logActivity = function (grainId, identityId, event) {
```

Comment:
Logically, you have two separate things in this file:

* you're declaring and implementing a function that is meant for reuse 
* you're executing some code that sets up some global state that is supposed to run exactly once

Their lifecycle is separate.  If you want to put the stuff that sets up the global state in a function, and move this whole file to `/imports` and just export two functions, and then import and call the function that sets up the global state from somewhere in `shell/server/`, that would be reasonable.

Long-term, I believe the appropriate structure will be to put everything in `/imports` and have everything under `shell/server` just import other things.